numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1144,7 +1162,16 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
     PyDataMem_FREE(mps);
     if (out != NULL && out != obj) {
         Py_INCREF(out);
-        PyArray_ResolveWritebackIfCopy(obj);
+        int success;
+        if (is_newarr) {
+            success = PyArray_CopyAnyInto(out, obj);
+        }
+        else {
+            success = PyArray_ResolveWritebackIfCopy(obj);
+        }
```

Comment:
And thus, you never need this branch here either.

Reply:
This as well.